include/seqan3/search/kmer_index/shape_iterator.hpp

```diff
@@ -0,0 +1,497 @@
+     */
+    void hash_reverse(difference_type const skip)
+    {
+        std::advance(text_left, -skip);
+        hash_full();
+    }
+
+    //!\brief Calculates a hash value by explicitly looking at each position.
+    void hash_full()
+    {
+        text_right = text_left;
+        hash_value = shape_[0] * to_rank(*text_right);
+
+        for (size_t i{1}; i < std::ranges::size(shape_); ++i)
+        {
+            std::advance(text_right, 1);
+
+            hash_value *= shape_[i] ? sigma : 1;
+            hash_value += shape_[i] * to_rank(*text_right);
+        }
+    }
+
+    //!\brief Calculates the next hash value via rolling hash.
+    void hash_roll()
+    {
+        hash_value -= to_rank(*(text_left)) * roll_factor;
+
+        std::advance(text_left,  1);
+        std::advance(text_right, 1);
+
+        hash_value *= sigma;
+        hash_value += to_rank(*(text_right));
+    }
+
+    /*!\brief Calculates the previous hash value via rolling hash.
+     * \attention This function is only avaible if it_t models std::BidirectionalIterator.
+     */
+    void hash_roll_back()
+        //!\cond
+        requires std::BidirectionalIterator<it_t>
+        //!\endcond
+    {
+        hash_value -= to_rank(*(text_right));
+
+        std::advance(text_left,  -1);
+        std::advance(text_right, -1);
+
+        hash_value /= sigma;
+        hash_value += to_rank(*(text_left)) * roll_factor;
+    }
+
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    shape_iterator()                                   = default; //!< Defaulted.
+    shape_iterator(shape_iterator const &)             = default; //!< Defaulted.
+    shape_iterator(shape_iterator &&)                  = default; //!< Defaulted.
+    shape_iterator & operator=(shape_iterator const &) = default; //!< Defaulted.
+    shape_iterator & operator=(shape_iterator &&)      = default; //!< Defaulted.
+    ~shape_iterator()                                  = default; //!< Defaulted.
+
+    /*!\brief Construct from a given iterator on the text and a seqan3::shape.
+    * /param[in] it_start Iterator pointing to the first position of the text.
+    * /param[in] s_ seqan3::shape
+    *
+    * ### Complexity
+    *
+    * Linear in size of shape.
+    */
+    shape_iterator(it_t it_start, shape s_) : shape_(s_),
+                                              text_start(it_start),
+                                              text_left(it_start),
+                                              text_right(it_start)
+    {
+        assert(std::ranges::size(shape_) > 0);
+        ungapped_shape = std::all_of(std::ranges::begin(shape_),
+                                     std::next(std::ranges::begin(shape_), std::ranges::size(shape_)),
+                                     [](bool b) { return b; } );
+
+        roll_factor = std::pow(sigma, std::ranges::size(shape_) - 1);
+
+        hash_full();
+    };
+    //!\}
+
+    //!\anchor shape_iterator_comparison
```

Comment:
What's anchor doing :eyes: ?

Reply:
I can do a `\ref` to an `\anchor`. But not to a `\name`.